### ISSUE TITLE
Problems with \forall and JMLCollections 

### ISSUE THREAD

**Issue by jhoenicke** [OpenJML/OpenJML#524]
Test classes under http://www.informatik.uni-freiburg.de/~hoenicke/fm4j/heap-example-1.tar.gz
I'm still struggling with my PriorityQueue example.  The problem now is:

```
Internal JML bug - please report.  BuildOpenJML-0.8.13-20170508
java.lang.NullPointerException
        at com.sun.tools.javac.comp.Lower.visitIterableForeachLoop(Lower.java:3486)
        at com.sun.tools.javac.comp.Lower.visitForeachLoop(Lower.java:3378)
        at com.sun.tools.javac.tree.JCTree$JCEnhancedForLoop.accept(JCTree.java:1035)
        at org.jmlspecs.openjml.JmlTree$JmlEnhancedForLoop.accept(JmlTree.java:1558)
```

I debugged a bit and the problem is that the \forall quantifier (in the spec of PriorityQueue) is translated to a for-loop, but JMLObjectBag does not implement Iterable.

**Comment by davidcok** [comment]
Working on it — it is a bit more than the Iterable issue.

**Comment by davidcok** [comment]
Fixed for the next release (0.8.15).
There still remains a fair bit of work on the model classes - they need to be made generic and many more of them declared Iterable.
